frontend/probloom/src/App.tsx

```diff
@@ -1,62 +1,92 @@
+import Footer from './components/Layout/Footer';
+import NotFound from './components/NotFound/NotFound';
+import AppHeader from './containers/AppHeader/AppHeader';
 import ProblemSetCreate from './containers/ProblemSet/ProblemSetCreate/ProblemSetCreate';
 import ProblemSetDetail from './containers/ProblemSet/ProblemSetDetail/ProblemSetDetail';
 import ProblemSetEdit from './containers/ProblemSet/ProblemSetEdit/ProblemSetEdit';
-import ProblemSetSolve from './containers/ProblemSet/ProblemSetSolve/ProblemSetSolve';
 import ProblemSetExplanationDetail from './containers/ProblemSet/ProblemSetExplanation/ProblemSetExplanationDetail';
-import NotFound from './components/NotFound/NotFound';
-import AppHeader from './containers/AppHeader/AppHeader';
-import Footer from './components/Layout/Footer';
+import ProblemSetSearch from './containers/ProblemSet/ProblemSetSearch/ProblemSetSearch';
+import ProblemSetSolve from './containers/ProblemSet/ProblemSetSolve/ProblemSetSolve';
+import Profile from './containers/Profile/Profile';
+import SignUp from './containers/SignUp/SignUp';
+import Welcome from './containers/Welcome/Welcome';
+import { getCurrentUser } from './store/actions/userActions';
+import { AppDispatch, RootState } from './store/store';
 
-function App(props: { history: History }) {
-  return (
-    <div className="App" style={{ paddingTop: '4rem' }}>
-      <ConnectedRouter history={props.history}>
-        <AppHeader />
-        <Switch>
-          <Route
-            path="/signin/"
-            exact
-            render={() => <Welcome history={props.history} logo="ProbLoom" />}
-          />
-          <Route
-            path="/signup/"
-            exact
-            render={() => <SignUp history={props.history} />}
-          />
-          <Redirect exact from="/" to="signin/" />
-          <Route path="/user/:id/:active/" exact component={Profile} />
-          <Redirect from="/user/:id/" to="/user/:id/summary/" exact />
-          <Route path="/problem/search/" exact component={ProblemSetSearch} />
-          <Route path="/problem/create/" exact component={ProblemSetCreate} />
-          <Route
-            path="/problem/:id/detail/"
-            exact
-            component={ProblemSetDetail}
-          />
-          <Route path="/problem/:id/solve/" exact component={ProblemSetSolve} />
-          <Route path="/problem/:id/edit/" exact component={ProblemSetEdit} />
-          <Route
-            path="/problem/:id/explanation/"
-            exact
-            component={ProblemSetExplanationDetail}
-          />
-          <Route component={NotFound} />
-        </Switch>
-        <Footer />
-      </ConnectedRouter>
-    </div>
-  );
+export interface AppProps extends PropsFromRedux {
+  history: History;
 }
 
-export default App;
+class App extends Component<AppProps> {
+  render() {
+    if (this.props.selectedUser === null) {
+      this.props.handleNoUser();
+      return null;
+    }
```

Comment:
This commit changes `App` component so that it fetches user information from server. As a result, the user is not logged out when reloading the page.